### PR TITLE
Improve error messaging when trying to attach a service connected to another account

### LINKED ISSUE #209: Error messages for attaching services aren't very helpful
If a user attaches their Twitter account to their Retrospring account then tries to do the same on another one, they will get a very unhelpful error message:

image

### PR DESCRIPTION
**Pull request by raccube**
Closes #209

### CODE REVIEW

File: app/controllers/services_controller.rb [hunk 1]
```diff
@@ -9,11 +9,16 @@ def index
...
 
-    if current_user.services << service
+    if service.save
       flash[:success] = t('flash.service.create.success')
     else
-      flash[:error] = t('flash.service.create.error')
+      if service.errors.details.has_key?(:uid) && service.errors.details[:uid].any? { |err| err[:error] == :taken }
+        flash[:error] = "The #{service.type.split('::').last.titleize} account you are trying to connect is already connected to another #{APP_CONFIG['site_name']} account."
+      else
+        flash[:error] = t('flash.service.create.error') + service.errors.inspect
```
Comment (nilsding) [inline app/controllers/services_controller.rb]:
```suggestion
        flash[:error] = t('flash.service.create.error')
```

**Review by nilsding — CHANGES_REQUESTED** [review]
Please add some tests

File: app/controllers/services_controller.rb [hunk 2]
```diff
@@ -9,11 +9,16 @@ def index
...
     service = Service.initialize_from_omniauth( omniauth_hash )
+    service.user = current_user
 
-    if current_user.services << service
+    if service.save
       flash[:success] = t('flash.service.create.success')
     else
-      flash[:error] = t('flash.service.create.error')
+      if service.errors.details.has_key?(:uid) && service.errors.details[:uid].any? { |err| err[:error] == :taken }
+        flash[:error] = "The #{service.type.split('::').last.titleize} account you are trying to connect is already connected to another #{APP_CONFIG['site_name']} account."
```
Comment (nilsding) [inline app/controllers/services_controller.rb]: maybe we should guide the user to our currently used support channel (the @Retrospring twitter) so one of us can sort this out

File: spec/controllers/services_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    after do
+      OmniAuth.config.mock_auth[:twitter] = nil
+    end
+
+    subject { get :create, params: { provider: 'twitter' } }
+
+    context 'no services connected' do
+    it 'creates a service integration' do
+      expect { subject }.to change { Service.count }.by(1)
+    end
```
Comment (nilsding) [inline spec/controllers/services_controller_spec.rb]: Indentation, please

**Review by nilsding — APPROVED** [review]

File: spec/controllers/services_controller_spec.rb (hunk 3 above)
Comment (raccube) [reply spec/controllers/services_controller_spec.rb]: 9ce1fc0c

File: app/controllers/services_controller.rb (hunk 2 above)
Comment (raccube) [reply app/controllers/services_controller.rb]: 9ce1fc0c

**Pull request merged**